### ISSUE TITLE
Adding requirements.txt

### ISSUE THREAD

**Issue by dgw** [dlenski/tetherback#37]
If a package contains a `requirements.txt` file, pip can automatically install python dependencies. With reference to #34, it would make the installation process slightly easier, even if tetherback itself is not in PyPI.

The stated requirements are progressbar2 and tabulate, but unknown which versions of each. @dlenski I'm happy to create and test a requirements file tonight if you have any hints what minimum (and/or maximum) versions of those dependencies work.

**Comment by dlenski** [comment]
My understanding of Python packaging is way out of date and spotty, but I thought [that the `install_requires` bit](https://github.com/dlenski/tetherback/blob/master/setup.py#L42) was supposed to specify this:

``` python
      install_requires=[ 'progressbar2>=3.6', 'tabulate' ],
```

I could've sworn there was an option to `pip` to install the dependencies as well, but can't find it at the moment.

**Comment by dgw** [comment]
I'm not even sure that the documentation is correct. I just read through a few docs on the file vs. setup parameter, and they indicate that the text file isn't used unless pip is given the `-r` argument. Yet, I `pip install` packages with only `requirements.txt` all the time, without that argument, and the dependencies are installed.

I might have to look into it more. It's true that I usually get Python 2.7 on servers, so even though pip should be consistent across versions it's possible that the behavior actually differs between py2 and py3.

**Comment by dlenski** [comment]
Yeah, I found the documentation frustratingly vague and inconsistent. Does your `pip3` _not_ seem to use the `install_requires` argument?

If you want to submit a PR with the correct `requirements.txt`, I'm all for it.

`progressbar2>=3.6` (cf. #12 where we figured out that this was the minimum version, as well as #20 and [python-progressbar issue #76](https://github.com/WoLpH/python-progressbar/issues/76) where there there was a small bug in `progressbar2`)
`tabulate` (not sure what version is required, because I haven't got any reports of one that's so old that it doesn't work :+1:)

**Comment by dgw** [comment]
tl;dr: `requirements.txt` is not necessary.

After sorting a Homebrew/Python3 issue that left me with a missing `pip3`, I managed to test this by:
- `pip3 uninstall tetherback tabulate progressbar2`
- `pip3 install https://github.com/dlenski/tetherback/archive/HEAD.zip`

Result:

```
Collecting https://github.com/dlenski/tetherback/archive/HEAD.zip
  Downloading https://github.com/dlenski/tetherback/archive/HEAD.zip
     \ 20kB 67kB/s
Collecting progressbar2>=3.6 (from tetherback==0.9.1)
  Downloading progressbar2-3.10.1-py2.py3-none-any.whl
Collecting tabulate (from tetherback==0.9.1)
Requirement already satisfied: python-utils>=2.0.0 in /usr/local/lib/python3.5/site-packages (from progressbar2>=3.6->tetherback==0.9.1)
Requirement already satisfied: six in /usr/local/lib/python3.5/site-packages (from python-utils>=2.0.0->progressbar2>=3.6->tetherback==0.9.1)
Installing collected packages: progressbar2, tabulate, tetherback
  Running setup.py install for tetherback ... done
Successfully installed progressbar2-3.10.1 tabulate-0.7.5 tetherback-0.9.1
```

So it seems that `pip3` does handle the `install_requires` property as expected. I'm not sure if that's always been the case, because upon initially installing tetherback I just manually installed the dependencies before tetherback itself (following the readme).

What I might do instead is rewrite the readme to indicate that the dependencies are needed, but not to indicate that the user must install them manually (which is the impression I get from the current wording). If you'd still like me to tack on a `requirements.txt` I'm happy to do so, but since it's apparently not required that would just add a second place to keep dependencies up-to-date.

**Comment by dlenski** [comment]
Thanks @dgw. I'm with you on not duplicating the requirements unless necessary.

Do you have any suggestions for how to word the requirements?

**Comment by dgw** [comment]
@dlenski I went ahead and took a rewrite pass in #38. End of the night for me, but I look forward to seeing your thoughts there tomorrow.

Closing this issue as invalid/unnecessary.
